.github/actions/akv-secret/index.js

```diff
@@ -0,0 +1,78 @@
+const core = require('@actions/core');
+const exec = require('@actions/exec');
+const fs = require('fs');
+
+(async () => {
+  try {
+    // Get inputs
+    const vaultName = process.env.INPUT_VAULT;
+    const secretsInput = process.env.INPUT_SECRETS;
+
+    // Parse and normalize secret mappings
+    const secretMappings = secretsInput
+      .split(/[\n,]+/)
+      .map((entry) => entry.trim())
+      .filter((entry) => entry)
+      .map((entry) => {
+        const [input, encoding, output] = entry.split(/(\S+)?>/).map((part) => part?.trim());
+        return { input, encoding, output: output || `\$output:${input}` }; // Default output to $output:input if not specified
+      });
+
+    if (secretMappings.length === 0) {
+      throw new Error('No secrets provided.');
+    }
+
+    // Fetch secrets from Azure Key Vault
+    for (const { input: secretName, encoding, output } of secretMappings) {
+      let secretValue = '';
+
+      const args = encoding
+        ? ['--encoding', encoding, '--file', '/dev/stdout']
+        : ['--query', 'value', '--output', 'tsv'];
+      await exec.exec('az', [
+        'keyvault',
+        'secret',
+        encoding ? 'download' : 'show',
+        '--vault-name',
+        vaultName,
+        '--name',
+        secretName,
+        ...args
+      ], {
+        listeners: {
+          stdout: (data) => {
+            secretValue += data.toString();
+          },
+        },
+      });
+
+      secretValue = secretValue.trim();
+
+      // Mask the secret value in logs
+      core.setSecret(secretValue);
+
+      if (output.startsWith('$env:')) {
+        // Environment variable
+        const envVarName = output.replace('$env:', '').trim();
+        fs.appendFileSync(process.env.GITHUB_ENV, `${envVarName}=${secretValue}\n`);
```

Comment:
I believe [this](https://github.com/actions/toolkit/tree/main/packages/core#exporting-variables) is the canonical way to do that:

```suggestion
        core.exportVariable(envVarName, secretValue);
```

That way, all kinds of corner cases are handled correctly, including multi-line values (which require some `<<<EOF` shenanigans).